### DISCUSSION TITLE
Swedish Library Notes

### DISCUSSION THREAD

**Discussion by tajmone** [alan-if/alan-i18n#26] [New Locales]
@thoni56, I've just discovered that at the [IF Archive] there's a Swedish translation of the Inform 6 library (to be used with Inform 6.31 and standard library 6/11) written by Fredrik Ramsberg. Here's a direct link to the Zip file:

- [swedish_inform.zip]

When I began translating the ALAN StdLib to Italian I've found it very useful to study how various other Italian libraries were implemented; I've learned from them how they would handle various grammar issues, how the main verbs were translated, and how the core messages looked like.

Hence, I though the above link might be useful for your translation work (I'm not aware of any other IF library in Swedish).

[IF Archive]: https://www.ifarchive.org
[swedish_inform.zip]:  https://www.ifarchive.org/if-archive/infocom/compilers/inform6/library/translations/swedish_inform.zip

**Comment by tajmone** [comment]
## Inform 6 Swedish

> @thoni56 (in #33): Actually, I know of no libraries for IF in Swedish (not that I'm all that into the eco-systems of the "other" tools). I think the number of works of IF in Swedish is also relatively low partly because we are (or consider ourselves to be) fairly good at English.

At the IFDB I've [found some adventures in Swedish][ifdb sv], but unfortunately they are not all tagged appropriately, so it's difficult to know how many there are exactly. There must surely be an Inform6 library in Swedish since I've came across Swedish games made with Inform:

- [Äventyr](https://ifdb.org/viewgame?id=ie63u67rmsn15ca5)
- [Pangnyheten](https://ifdb.org/viewgame?id=f0r56g6r8tgj3w8n)

The [IF Archive] has a Zip of the Swedish Inform6 library, [`swedish_inform.zip`][swedish_inform.zip]; but I have no idea of its current state. Also mentioned at Inform website:

- https://inform-fiction.org/translations/swedish.html

and seems to have its own homepage too:

- http://www.microheaven.com/svenska/skriva.shtml

From that page, it looks like there actually two different i6 Swedish libraries, _Tidsformer_ and a newer version. But not knowing Swedish I can't gain much insight into it.





[ifdb sv]:https://ifdb.org/search?searchfor=swedish&searchgo=Search+Games

[IF Archive]: https://www.ifarchive.org/indexes/if-archive/infocom/compilers/inform6/library/translations/

[swedish_inform.zip]: https://www.ifarchive.org/if-archive/infocom/compilers/inform6/library/translations/swedish_inform.zip


[Stugan cover]: https://www.mobygames.com/images/covers/l/178278-stugan-dos-front-cover.jpg
[Stugan screenshot]: https://www.mobygames.com/images/shots/l/590820-stugan-dos-screenshot-stugan-intro-screen.png

**Reply by tajmone** [reply]
### Inform 6 Swedish Sources in Wiki

@thoni56, I've added to the Wiki a selection of source files from the Swedish Inform 6 library, converted to UTF-8, which we can use as references for the Foundation Swedish library:

https://github.com/alan-if/alan-i18n/wiki/i18-Porting-References#swedish-inform

The i6 source contain useful stuff, e.g. the various GNA tables and their code, as well as the various verbs/commands translations and their response messages, which can be really useful.

**Reply by thoni56** [reply]
Thanks, @tajmone, that's *really* interesting! Did you pick up on "Äventyr" being a translation of Adventure? And it was rated as recently as in April this year by a blind player? So perhaps your endevour to have IF make gaming more accessible to blind people are panning out.

I'll have a more detailed look at those when I get enough time to get back to alani18n. And it will be very valuable for thinking about how those things can influence the Alan language. GNA is easier to grasp for me if they are in Swedish I think.

**Reply by tajmone** [reply]
> Did you pick up on "Äventyr" being a translation of Adventure?

yes, that was intuitive enough, as well as _svenska_ being Swedish, but not much more.

>  And it was rated as recently as in April this year by a blind player? 

No, I didn't manage to grasp that, but it's very interesting. I did notice that the project seems well polished and organized.

> GNA is easier to grasp for me if they are in Swedish I think.

I think that the _Inform Design Manual_ remains the best reference for GNA, parsing, etc., for any language, for it was written with internationalization in mind (not just for Inform 6 either).
